raiden-dapp/src/components/dialogs/UdcDepositDialog.vue

```diff
@@ -90,32 +108,53 @@ import { RaidenError } from 'raiden-ts';
   },
 })
 export default class UdcDepositDialog extends Vue {
-  amount: string = '10';
+  mainnet!: boolean;
+  uniswapURL: string = '';
+  udcToken!: Token;
+  utilityTokenAmount: string = '';
+  step: 'mint' | 'approve' | 'deposit' | '' = '';
   loading: boolean = false;
   error: Error | RaidenError | null = null;
 
-  step: 'mint' | 'approve' | 'deposit' | '' = '';
-
   @Prop({ required: true, type: Boolean })
   visible!: boolean;
 
-  mainnet!: boolean;
-  udcToken!: Token;
+  @Emit()
+  cancel() {
+    this.error = null;
+  }
 
   get valid(): boolean {
-    return /^[1-9]\d*$/.test(this.amount);
+    return /^[1-9]\d*$/.test(this.utilityTokenAmount);
   }
 
-  @Emit()
-  cancel() {
-    this.error = null;
+  async mounted() {
+    if (this.mainnet) {
+      const mainAccountAddress =
+        (await this.$raiden.getMainAccount()) ??
```

Comment:
These are the functions in the ```raiden-service.ts```:
```ts
async getMainAccount(): Promise<string | undefined> {
   return this.raiden.mainAddress;
}

async getAccount(): Promise<string> {
  return this.raiden.address;
}
```

Two questions:
1. So, I don't need to make them async even though they are in ```raiden-service.ts```
2. ```const mainAccountAddress = this.$raiden.mainAccount ?? this.$raiden.account;```, does not work because I believe they are not actual getters in ```raiden-service.ts```, or?

Reply:
Note: we had a call to discuss this PR.

We will need to adust the raiden service and all usages of the "getters".

Reply:
I created a separate issue for this #1983 since it seems to be more involved than we initially thought.